custom/enikshay/utils.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+from collections import defaultdict
+
+from django.utils.dateparse import parse_datetime, parse_date
+
+from corehq.apps.fixtures.models import FixtureDataItem, FixtureDataType
+from corehq.form_processor.exceptions import LedgerValueNotFound
+from corehq.form_processor.interfaces.dbaccessors import LedgerAccessors
+from corehq.util.quickcache import quickcache
+from casexml.apps.stock.mock import (
+    Balance,
+    Entry,
+)
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.users.util import SYSTEM_USER_ID
+
+
+def get_adherence_cases_by_date(adherence_cases):
+    adherence_cases_by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherence_date = parse_date(case['adherence_date']) or parse_datetime(case['adherence_date']).date()
+        adherence_cases_by_date[adherence_date].append(case)
+    return adherence_cases_by_date
+
+
+def get_episode_adherence_ledger(domain, episode_case_id, entry_id):
+    """
+    :param domain: domain name
+    :param episode_case_id: episode case id for adherence
+    :param entry_id: example date_2017-12-09
+    """
+    ledger_accessor = LedgerAccessors(domain)
+    try:
+        return ledger_accessor.get_ledger_value(episode_case_id, "adherence", entry_id)
+    except LedgerValueNotFound:
+        return None
+
+
+@quickcache(['domain'], memoize_timeout=7 * 24 * 60 * 60, timeout=7 * 24 * 60 * 60)
+def get_id_of_fixture_tagged_adherence_ledger_values(domain):
```

Comment:
nit: this name is very long (almost as long as the function itself)! could rename it as `get_adherence_fixture_id` or something.